tests/unit/parsers/test.manifestjson.js

```diff
@@ -2953,6 +2954,52 @@ describe('ManifestJSONParser', () => {
     });
   });
 
+  describe('sitepermission', () => {
+    it('supports simple valid sitepermission', () => {
+      const linter = new Linter({ _: ['bar'] });
+      const json = validSitePermissionManifestJSON();
+      const manifestJSONParser = new ManifestJSONParser(json, linter.collector);
+      expect(manifestJSONParser.isValid).toEqual(true);
+    });
+
+    it('detects invalid sitepermission (more than one install_origin)', () => {
+      const linter = new Linter({ _: ['bar'] });
+      const json = validSitePermissionManifestJSON({
+        install_origins: ['https://website1.com', 'https://website2.com'],
+      });
+      const manifestJSONParser = new ManifestJSONParser(json, linter.collector);
+      expect(linter.collector.errors).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            code: messages.JSON_INVALID.code,
+            message: '"/install_origins" should NOT have more than 1 items',
+            dataPath: '/install_origins',
+          }),
+        ])
+      );
+      expect(manifestJSONParser.isValid).toEqual(false);
+    });
+
+    it('detects invalid sitepermission (invalid site_permissions value)', () => {
+      const linter = new Linter({ _: ['bar'] });
+      const json = validSitePermissionManifestJSON({
+        site_permissions: ['midi', 'not_a_valid_webapi_name'],
+      });
+      const manifestJSONParser = new ManifestJSONParser(json, linter.collector);
+      expect(linter.collector.errors).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            code: messages.JSON_INVALID.code,
+            message:
+              '"/site_permissions/1" is not a valid key or has invalid extra properties',
```

Comment:
@willdurand @Rob--W 

To be fair this isn't exactly the same kind of validation error message we use for invalid permissions, for those we tweak the message as collected by ajv from the manifest parser class. 

We could do the same for the site_permissions but I'm not sure it would be really valuable given that this addon type submissions should be actually go through the usual submission flow used for the "real" webextensions (and so I opted to not do that, at least not until we decided that there would be value in doing that).